packages/nextjs/src/config/nextConfigToWebpackPluginConfig.ts

```diff
@@ -0,0 +1,128 @@
+  nextConfig: NextConfigObject,
+  sentryWebpackPluginOptions: Partial<SentryWebpackPluginOptions>,
+): Partial<SentryWebpackPluginOptions> {
+  return includeNextjsProps(nextConfig, sentryWebpackPluginOptions, PROPS_INCLUDER_MAP, SUPPORTED_NEXTJS_PROPERTIES);
+}
+
+/**
+ * Creates a new Sentry Webpack Plugin config from the given one, and applying the corresponding
+ * modifications to the given next properties. If more than one option generates the same
+ * properties, the values generated last will override previous ones.
+ *
+ * @param nextConfig User's Next.js config.
+ * @param sentryWebpackPluginOptions User's Sentry Webapck Plugin config.
+ * @param nextProps Next.js config's properties that should modify webpack plugin properties.
+ * @returns New Sentry Webpack Plugin config.
+ */
+export function includeNextjsProps(
+  nextConfig: NextConfigObject,
+  sentryWebpackPluginOptions: Partial<SentryWebpackPluginOptions>,
+  propsIncluderMap: Record<string, PropIncluderFn>,
+  nextProps: string[],
+): Partial<SentryWebpackPluginOptions> {
+  // @ts-ignore '__spreadArray' import from tslib, ts(2343)
+  const propsToInclude = [...new Set(nextProps)];
+  return (
+    propsToInclude
+      // Types are not strict enought to ensure there's a function in the map
+      .filter(prop => propsIncluderMap[prop])
+      .map(prop => propsIncluderMap[prop](nextConfig, sentryWebpackPluginOptions))
+      .reduce((prev, current) => ({ ...prev, ...current }), {})
+  );
+}
+
+/**
+ * Creates a new Sentry Webpack Plugin config with the `distDir` option from Next.js config
+ * in the `include` property.
+ *
+ * If no `distDir` is provided, the Webpack Plugin config doesn't change.
+ * If no `include` has been defined defined, the `distDir` value is assigned.
+ * The `distDir` directory is merged to the directories in `include`, if defined.
+ * Duplicated paths are removed while merging.
+ *
+ * @param nextConfig User's Next.js config
+ * @param sentryWebpackPluginOptions User's Sentry Webpack Plugin config
+ * @returns New Sentry Webpack Plugin config
+ */
+export function includeDistDir(
+  nextConfig: NextConfigObject,
+  sentryWebpackPluginOptions: Partial<SentryWebpackPluginOptions>,
+): Partial<SentryWebpackPluginOptions> {
+  if (!nextConfig.distDir) {
+    return { ...sentryWebpackPluginOptions };
+  }
+  // It's assumed `distDir` is a string as that's what Next.js is expecting. If it's not, Next.js itself will complain
+  const usersInclude = sentryWebpackPluginOptions.include;
+
+  let sourcesToInclude;
+  if (typeof usersInclude === 'undefined') {
+    sourcesToInclude = nextConfig.distDir;
+  } else if (typeof usersInclude === 'string') {
+    sourcesToInclude = usersInclude === nextConfig.distDir ? usersInclude : [usersInclude, nextConfig.distDir];
+  } else if (Array.isArray(usersInclude)) {
+    // @ts-ignore '__spreadArray' import from tslib, ts(2343)
+    sourcesToInclude = [...new Set(usersInclude.concat(nextConfig.distDir))];
```

Comment:
```suggestion
    sourcesToInclude = Array.from(new Set([...usersInclude, nextConfig.distDir]));
```

Reply:
We can also just use a `.filter` instead of `Array.from` + `Set`, IMO makes intent clearer but up to you.